posthog/api/test/test_hog_function.py

```diff
@@ -0,0 +1,264 @@
+import json
+from unittest.mock import ANY
+
+from rest_framework import status
+
+from posthog.models.action.action import Action
+from posthog.test.base import APIBaseTest, ClickhouseTestMixin, QueryMatchingTest
+
+
+EXAMPLE_FULL = {
+    "name": "HogHook",
+    "hog": "fetch(inputs.url, {\n  'headers': inputs.headers,\n  'body': inputs.payload,\n  'method': inputs.method\n});",
+    "inputs_schema": [
+        {"key": "url", "type": "string", "label": "Webhook URL", "required": True},
+        {"key": "payload", "type": "json", "label": "JSON Payload", "required": True},
+        {
+            "key": "method",
+            "type": "choice",
+            "label": "HTTP Method",
+            "choices": [
+                {"label": "POST", "value": "POST"},
+                {"label": "PUT", "value": "PUT"},
+                {"label": "PATCH", "value": "PATCH"},
+                {"label": "GET", "value": "GET"},
+            ],
+            "required": True,
+        },
+        {"key": "headers", "type": "dictionary", "label": "Headers", "required": False},
+    ],
+    "inputs": {
+        "url": {
+            "value": "http://localhost:2080/0e02d917-563f-4050-9725-aad881b69937",
+        },
+        "method": {"value": "POST"},
+        "headers": {
+            "value": {"version": "v={event.properties.$lib_version}"},
+        },
```

Comment:
The bytecode is the thing yeah. Otherwise we would need an entire bytecode object next to it that mimics the structure. 
Also there might be something like "encrypted value" later.